model/dce-fd.cc

```diff
@@ -135,6 +136,61 @@ int dce_open (const char *path, int flags, ...)
   return fd;
 }
 
+int dce_posix_fallocate(int fd, off_t offset, off_t len)
+{
+  return 0;
+}
+
+int dce_openat (int dirfd, const char *pathname, int flags, ...) {
+  va_list vl;
+  va_start (vl, flags);
+
+  mode_t mode = 0;
+  if (flags & O_CREAT)
+    {
+      mode = va_arg (vl, mode_t);
+    }
+  va_end (vl);
+  
+  char c = *pathname;
+  if (c == '/' || dirfd == AT_FDCWD) {
+    return dce_open(pathname, flags, mode);
+  }
+  
+  Thread *current = Current ();
+  int realFd = getRealFd (dirfd, current);
+
+  struct stat finfo;
+  int error = fstat(realFd, &finfo);
+  if (error)
+  {
+    return -1;
+  }
+
+  std::stringstream command_cp;    
+  command_cp << "cd files-" << UtilsGetNodeId () << "; find . -inum " << finfo.st_ino;
```

Comment:
it would be nice to add comment lines to describe what the followings are trying to accomplish.

Reply:
I've added a commit with some comments in this function. I know it isn't a good implementation, but it does the work.